MediaAdminBundle/Controller/Api/MediaController.php

```diff
@@ -113,13 +113,22 @@ public function uploadAction($folderId, Request $request)
 
         if ($uploadedFile && $filename = $saveMediaManager->getFilenameFromChunks($uploadedFile)) {
 
-            $media = $this->container->get('open_orchestra_media.manager.media')
-                ->createMediaFromUploadedFile($uploadedFile, $filename, $folderId);
+            if ($saveMediaManager->isFileAllowed($filename)) {
+                $media = $this->container->get('open_orchestra_media.manager.media')
+                    ->createMediaFromUploadedFile($uploadedFile, $filename, $folderId);
+
+                return $this->get('open_orchestra_api.transformer_manager')
+                    ->get('media')->transform($media);
+            } else {
+                $translator = $this->container->get('translator');
 
-            return $this->get('open_orchestra_api.transformer_manager')
-                ->get('media')->transform($media);
+                return new Response(
```

Comment:
I tried to throws exceptions but it breaks the flow.js treatment of the return, instead returning a 403 or 202 response goes fine.
The Api\Serialize annotation is only read if the KernelEvents::VIEW is fired. This is only the case when a non Response value is return by a controller. So when I return a custom Response in this controller, the serialize mechanism is even not called and that's why it works.